crates/table/src/var_len.rs

```diff
@@ -546,12 +533,14 @@ impl<'offsets, 'row> Iterator for AlignedVarLenOffsetsIterMut<'offsets, 'row> {
             //         mean that `row` is always 2-byte aligned, so this will be too,
             //         and that `row` is large enough for all the `offsets`,
             //         so this `add` is always in-bounds.
-            let elt_ptr: *mut MaybeUninit<VarLenRef> =
+            let elt_ptr: *mut VarLenRef =
                 unsafe { self.row.add(curr_offset_idx * mem::align_of::<VarLenRef>()).cast() };
 
             // SAFETY: `elt_ptr` is aligned and inbounds.
-            //         `MaybeUninit<VarLenRef>` has no value restrictions,
-            //         so it's safe to create an `&mut` to `uninit` or garbage.
+            //         Any sequence of non-`poison` bytes is valid at `VarLenRef`,
```

Comment:
Changing to "Any pattern of init bytes is valid [...]", because this has never been a complete description (e.g. alignment is not included), and I making it a complete description would be undesirable noise.